### PR TITLE
[DEM-278] Change feature example from aside to centered modal.

### PR DESCRIPTION
**Pull request by vietnogi**
ping @appirio-tech/demandside

### CODE REVIEW

File: src/client/app/submit-work/features/scripts/features.controller.js [hunk 1]
```diff
@@ -16,12 +16,10 @@
...
     vm.newFeatureExplanation = '';
 
-    vm.showExample = function (example) {
-      $rootScope.$emit('submit-work-show-example', example);
-    };
+    $scope.showExample       = false;
 
-    vm.hideExample = function () {
-      $rootScope.$emit('submit-work-hide-example');
+    $scope.clickExample = function () {
```
Comment (tladendo) [inline src/client/app/submit-work/features/scripts/features.controller.js]: publicly available variables and functions on `$scope`/`vm` should be at the top, while functions should be at the bottom of the controller (and hoisted up to the top). as per the styleguide

Comment (tladendo) [inline src/client/app/submit-work/features/scripts/features.controller.js]: also, let's have a discussion about `$scope` vs. `vm` tomorrow

File: src/client/app/layout/scripts/modal.directive.js [hunk 2]
```diff
@@ -26,6 +17,20 @@
...
       };
 
+      var close = function () {
+        scope.show = false;
+        scope.$apply();
+      }
+
+      closeButton.prependTo(element).bind('click', close);
+
+      if (!overlay.length) {
```
Comment (tladendo) [inline src/client/app/layout/scripts/modal.directive.js]: to be clear, this means that we won't add another modal to the DOM until there are no others around, right?

**Comment by tladendo** [comment]
lgtm besides comments

File: src/client/app/submit-work/features/scripts/features.controller.js (hunk 1 above)
Comment (vietnogi) [inline src/client/app/submit-work/features/scripts/features.controller.js]: ok, lets also discuss the style guide of having all functions at the bottom

File: src/client/app/layout/scripts/modal.directive.js (hunk 2 above)
Comment (vietnogi) [inline src/client/app/layout/scripts/modal.directive.js]: correct

**Comment by vietnogi** [comment]
ping @tladendo  updated the code to use the vm style.

**Comment by tladendo** [comment]
lgtm

**Pull request merged**